subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -412,7 +415,8 @@ int aws_fota_mqtt_evt_handler(struct mqtt_client *const client,
 				return err;
 			}
 		}
-		return 0;
+		/* Don't consume the publish event */
+		return 1;
```

Comment:
It looks like the checks earlier (`if (err < 1)`) in the PUBLISH case can return with 0, even if the event should be deferred to another MQTT handler. I think the return value and documentation of `aws_fota_on_publish_evt` might need an update

Reply:
Maybe something more like these changes?